mood-analysis.rb

```diff
@@ -19,6 +19,52 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(sentence)
+  sentence.gsub(/[[:punct:]]/, "")
+end
+
+def happy_days(journal)
+  entries = 0
+  happy_count = 0
+  # wanted to do this with reduce but didn't get it figuerd out :/
+  journal.each do |entry|
+    happy_count += 1 if analyze_mood(strip_punctuation(entry)) == ":-)"
+    entries += 1
+    if happy_count > 3
+      break
+    end
+  end
+  if happy_count < 3
+    puts "In all #{entries} enterest, there were not 3 happy days! Sorry :-(."
+  else
+    puts "It takes #{entries} entries for 3 happy days to occur."
+  end
+end
+
+def overall_mood(journal)
+  entry_moods_frequency = {":-)" => 0, ":-(" => 0, ":-|" => 0}
+
+  journal.each do |entry|
+    mood = analyze_mood(strip_punctuation(entry))
+    case mood
+    when ":-)"
+      entry_moods_frequency[":-)"] += 1
+    when ":-("
+      entry_moods_frequency[":-("] +=1
+    when ":-|"
+      entry_moods_frequency[":-|"] +=1
+    end
+
+  end
+
+  most_common_mood = entry_moods_frequency.max_by do |mood, count|
+    count
+  end
+
+  puts "The most common mood is #{most_common_mood[0]}. #{entry_moods_frequency}"
+
+end
```

Comment:
I would rather this method only return the emoticon, and not a string with the emoticon burried inside. That would allows us to do more things with this information rather than limiting us to this specific message/usage.